File: app/src/main/java/com/gustafson/ctcandroid_new/ListViewFragment.java
package com.gustafson.ctcandroid_new;

import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
<<<<<<< Updated upstream
import android.widget.TextView;
=======

import org.json.JSONArray;
import org.json.JSONObject;
>>>>>>> Stashed changes

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.ArrayList;

public class ListViewFragment extends Fragment {
<<<<<<< Updated upstream
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstances){
        View rootview = inflater.inflate(R.layout.list_view_fragment, container, false);
        //((TextView) rootview.findViewById(R.id.fragment_1_textview)).setText("Fragment Number One");
        ArrayList<CTCEvents> eventsList = new ArrayList<CTCEvents>();

        //this is where data would likely be added 

        eventsList.add(new CTCEvents("Test name", "TayTow2", "right now", "420"));
        eventsList.add(new CTCEvents("Test name TOO", "Code that cares classroom", "uh", "soon"));
        eventsList.add(new CTCEvents("Test name", "TayTow2", "right now", "420"));
        eventsList.add(new CTCEvents("Test name TOO", "Code that cares classroom", "uh", "soon"));
        eventsList.add(new CTCEvents("Test name", "TayTow2", "right now", "420"));
        eventsList.add(new CTCEvents("Test name TOO", "Code that cares classroom", "uh", "soon"));
        eventsList.add(new CTCEvents("Test name", "TayTow2", "right now", "420"));
        eventsList.add(new CTCEvents("Test name TOO", "Code that cares classroom", "uh", "soon"));
        eventsList.add(new CTCEvents("Test name", "TayTow2", "right now", "420"));
        eventsList.add(new CTCEvents("Test name TOO", "Code that cares classroom", "uh", "soon"));
        eventsList.add(new CTCEvents("Test name", "TayTow2", "right now", "420"));
        eventsList.add(new CTCEvents("Test name TOO", "Code that cares classroom", "uh", "soon"));
        eventsList.add(new CTCEvents("Test name", "TayTow2", "right now", "420"));
        eventsList.add(new CTCEvents("Test name TOO", "Code that cares classroom", "uh", "soon"));
        ((ListView) rootview.findViewById(R.id.list_view)).setAdapter(new ListViewAdapter(getContext(), eventsList));
        return rootview;
=======
    private static String url = "https://us-central1-ctcios.cloudfunctions.net/getAllEvents";
    private ArrayList<CTCEvents> ctcEvents;
    ListView ls;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstances){
        final View rootview = inflater.inflate(R.layout.list_view_fragment, container, false);
        ctcEvents = new ArrayList<>();
        ls = (ListView) rootview.findViewById(R.id.list_view);
        new getJsonData().execute(url);
        //((ListView) rootview.findViewById(R.id.list_view)).setAdapter(new ListViewAdapter(getContext(), ctcEvents));
        return rootview;
    }

    private class getJsonData extends AsyncTask<String, Void, ArrayList<CTCEvents>> {
        @Override
        protected void onPreExecute(){

        }

        @Override
        protected ArrayList<CTCEvents> doInBackground(String... urlString){
            URL url;
            StringBuffer response = new StringBuffer();
            try {
                url = new URL(urlString[0]);
            } catch (MalformedURLException e){
                throw new IllegalArgumentException("invalid url");
            }
            HttpURLConnection conn = null;
            try {
                conn = (HttpURLConnection) url.openConnection();
                conn.setDoOutput(false);
                conn.setDoInput(true);
                conn.setUseCaches(false);
                conn.setRequestMethod("GET");
                conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8");
                int status = conn.getResponseCode();
                if (status != 200){
                    throw new IOException("Post failed with error code " + status);
                } else {
                    BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
                    String inputLine;
                    while ((inputLine = in.readLine()) != null) {
                        response.append(inputLine);
                    }
                    in.close();
                }
            } catch (Exception e){
                e.printStackTrace();
            } finally {
                if (conn != null){
                    conn.disconnect();
                }
            }
            return formatToJson(response.toString());
        }

        @Override
        protected void onPostExecute(ArrayList<CTCEvents> events){
            ls.setAdapter(new ListViewAdapter(getContext(), events));
        }

        private ArrayList<CTCEvents> formatToJson(String s){
            ArrayList<CTCEvents> eventsList = new ArrayList<>();
            try {
                JSONArray response = new JSONArray(s);
                for(int i = 0; i < response.length(); i++){
                    JSONObject json = response.getJSONObject(i);
                    CTCEvents CTCEvents = new CTCEvents();
                    CTCEvents.setTimeStamp(json.getInt("timeStamp"));
                    CTCEvents.setDisplayColor(json.getString("displayColor"));
                    CTCEvents.setDetail(json.getString("detail"));
                    CTCEvents.setDurationMinutes(json.getInt("durationMinutes"));
                    CTCEvents.setTitle(json.getString("title"));
                    CTCEvents.setLocation(json.getString("location"));
                    eventsList.add(CTCEvents);
                }
            } catch (Exception e){
                e.printStackTrace();
            }
            return eventsList;
        }
>>>>>>> Stashed changes
    }
}

File: app/src/main/java/com/example/codethatcaresandroid/ListViewFragment.kt
package com.example.codethatcaresandroid

import android.content.Context
import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import org.json.JSONArray
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL
import java.util.ArrayList

val events = ArrayList<CTCEvents>()
private const val url: String = "https://us-central1-ctcios.cloudfunctions.net/getAllEvents"
private lateinit var recycler: RecyclerView
private lateinit var viewAdapter: RecyclerView.Adapter<*>
private lateinit var viewManager: RecyclerView.LayoutManager

class ListViewFragment : Fragment(), RecyclerCallbackInterface {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView: View = inflater.inflate(R.layout.calendar_view_fragment, container, false)
        viewManager = LinearLayoutManager(context)
        viewAdapter = CalendarAdapter(events, this)
        recycler = rootView.findViewById<RecyclerView>(R.id.recycler_view).apply {
            setHasFixedSize(true)
            layoutManager = viewManager
            adapter = viewAdapter
        }
        getJsonData(context!!).execute(url)
        return rootView
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

    }

    override fun startActivityFromAdapter(intent: Intent) {
        startActivity(intent)
    }

    class getJsonData(val c: Context) : AsyncTask<String, Void, ArrayList<CTCEvents>>() {
        override fun onPreExecute() {

        }

        override fun doInBackground(vararg urlString: String): ArrayList<CTCEvents> {
            val url: URL
            val response = StringBuffer()
            try {
                url = URL(urlString[0])
            } catch (e: MalformedURLException) {
                throw IllegalArgumentException("invalid url")
            }

            var conn: HttpURLConnection? = null
            try {
                conn = url.openConnection() as HttpURLConnection
                conn.doOutput = false
                conn.doInput = true
                conn.useCaches = false
                conn.requestMethod = "GET"
                conn.setRequestProperty("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
                val status = conn.responseCode
                if (status != 200) {
                    throw IOException("Post failed with error code $status")
                } else {
                    val input = BufferedReader(InputStreamReader(conn.inputStream))
                    for(inputLine in input.lines()) {
                        response.append(inputLine)
                    }
                    input.close()
                }
            } catch (e: Exception) {
                e.printStackTrace()
            } finally {
                conn?.disconnect()
            }
            return formatToJson(response.toString())
        }

        override fun onPostExecute(eventsList: ArrayList<CTCEvents>) {
            events.clear()
            events.addAll(eventsList)
            viewAdapter.notifyDataSetChanged()
        }

        private fun formatToJson(s: String): ArrayList<CTCEvents> {
            val eventsList = ArrayList<CTCEvents>()
            try {
                val response = JSONArray(s)
                for (i in 0 until response.length()) {
                    val json = response.getJSONObject(i)
                    val ctcEvents = CTCEvents(json)
                    eventsList.add(ctcEvents)
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }

            return eventsList
        }
    }
}